### PR TITLE
Desword blocks at least one hit guaranteed every 30 seconds

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Every 30 seconds the desword blocks at hit guaranteed. It otherwise retains all other functions.

In part this is an answer to some of the problems brought up with the desword in this pr https://github.com/tgstation/tgstation/pull/65053 but I'd really rather not contribute to muddying that PR even further. This can easily be implemented irregardless of whether that PR is merged or not and in fact is worth considering separately as well.

## Why It's Good For The Game

This was something waltermelon suggested (but mostly in context with the armour pr). I actually think it's a good idea in of itself. While the justification for this idea was so that the desword wasn't 'basically esword but more damage' with regards to the DT system, I actually honestly think that is true right now as well. Desword really is just esword but with an extremely insignificant amount of additional damage and robbing you of an offhand to use.

waltermelon saying some stuff

The desword, while generally feared by players who probably get spooked by much older players who remember when it was far more formidable (it could stun you), isn't actually very good when you consider the math behind it, and is only really going to pose a threat to someone who isn't very experienced or who would probably lose to anyone with any weapon at all (like, even a toolbox or fire extinguisher). 

With an RNG system to protect you from melee hits, there is a 1 in 4 chance every hit from any baton that you simply eat shit and get knocked down. Some batons can be instant knockdown, so in those instances this can be basically end the fight then and there as your sword becomes a tiny hard to click sprite on the floor (and you've probably been shoved). Batons are the de facto melee weapon that matters, so we can't really rely on the desword to help us fight those.

The amount of hits to bring one target down is 3-4 (accounting for limb deviation). In the same time it took to get hit once, you needed to have landed up to 4 hits of your own, not considering; armour, shields they're using (riot shields block deswords at 50%), some other misc. protection, all of which will have increased this amount even further.

In comparison, esword brings someone into crit in roughly the same amount of hits (4-5). It only blocks about every other hit, but at half the price, it works just fine as a defense tool against someone who doesn't have a baton. And also has the same amount of armoir penetration. If I truly wanted someone dead with a melee weapon, but already resigned to the fact that I can't win in a baton joust, because who I am killing probably isn't able to produce one of those against me (sec, heads of staff) or I'm not going to enter melee before they're down (I'm using an ebow/disabler/any ranged weapon really), I'll use the esword. Also, it only takes up one hand. I can shoot AND block with an esword. That's flexibility I quite like. 

With this change, you can at least get one more block squeezed in before rolling the dice. Just that one block can make all the difference. And that really does make it a reasonable consideration for whether using this expensive flashlight is worth your time.

#### TL;DR: I'm getting really tired of people pissing their pants at the desword when it sucks immense ass in meaningful combat situations, so maybe I just wanna justify them staining those jeans of theirs and making people hesitate running into melee with the fella without backup. You know, the thing that we say should be the crew advantage.

Yep that's my explanation okay thanks for coming to my TEDtalk about the desword.

## Changelog

:cl:
balance: The double-bladed energy sword can guarantee a block every 30 seconds.
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
Or you could just remove the bullshit one hit win stun combat? Lol.

**Comment by necromanceranne** [comment]
> Or you could just remove the bullshit one hit win stun combat? Lol.

Cute
believe me I tried

**Comment by necromanceranne** [comment]
might be likely this gets closed in favour of just redoing all block entirely in this same fashion. Just an fyi.

**Comment by SkeletalElite** [comment]
30 seconds is pretty long, maybe a little shorter. Like 20ish?

File: code/game/objects/items/dualsaber.dm [hunk 1]
```diff
@@ -34,6 +34,10 @@
 	var/hacked = FALSE
 	var/list/possible_colors = list("red", "blue", "green", "purple")
 	var/wielded = FALSE // track wielded status on item
+	//Whether or not we are poised, AKA whether we can block an attack reliably or not.
+	var/offbalance = FALSE
+	var/poise_recory_time = 30 SECONDS //Our timer for our poise recovery
```
Comment (SuperNovaa41) [inline code/game/objects/items/dualsaber.dm:39]: Shouldn't this just be a define?

**Comment by Fikou** [comment]
dont we have a shielded component that acts just like this

**Comment by necromanceranne** [comment]
> dont we have a shielded component that acts just like this

you and I worked out the same thing. I'm probably closing this in the morning and having a look into that.

**Comment by SgtHunk** [comment]
honestly I'd personally prefer a system where the desword (and esword) is actually threatening in melee combat rather than "haha rng chance for you to win this fight if you have a baton"
I'd love if it could saw off arms of people who shoved you as a "fuck you" to people who use shove combat as a crutch lol

**Comment by ATH1909** [comment]
poise bois

**Comment by ATH1909** [comment]
> honestly I'd personally prefer a system where the desword (and esword) is actually threatening in melee combat rather than "haha rng chance for you to win this fight if you have a baton" I'd love if it could saw off arms of people who shoved you as a "fuck you" to people who use shove combat as a crutch lol

imo, the real problem there is that batons can win the fight with just one hit getting through

but maybe that's just me

**Comment by Imaginos16** [comment]
Since the PR itself doesn't seem to touch on the sprite part, what are these for @necromanceranne?
image

**Comment by necromanceranne** [comment]
I explained the sprites over discord and what the funniness is with the component.

Also, I'm gonna close this PR for now, I'll get to the bigger stuff later.
